File: 백준/Silver/28278. 스택 2/스택 2.java
import java.util.*;
import java.io.*;
import java.nio.*;

class Main{
    static StringTokenizer st;

    public static int getToken(){
        return Integer.parseInt(st.nextToken());
    }
    public static int getToken(String s){
        return Integer.parseInt(s);
    }

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder answer = new StringBuilder();
        Stack <Integer> stack = new Stack<>();
        int n = getToken(br.readLine());

        while(n-- > 0){
            st = new StringTokenizer(br.readLine());
            int x = getToken();

            if(x == 1){
                stack.push(getToken());
            } else if (x == 2){
                answer.append(stack.isEmpty() ? -1 : stack.pop()).append('\n');
            } else if (x == 3){
                answer.append(stack.size()).append('\n');
            } else if (x == 4){
                answer.append(stack.isEmpty() ? 1 : 0).append('\n');
            } else {
                answer.append(stack.isEmpty() ? -1 : stack.peek()).append('\n');
            }
        }

        System.out.println(answer);
    }
}

File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

n = int(sys.stdin.readline())
stack = []

for _ in range(n):
    com = sys.stdin.readline().split()
    c = int(com[0])
    if(c == 1):
        stack.append(int(com[1]))
    elif c == 2:
        if stack:
            print(stack.pop())
        else :
            print(-1)
    elif c == 3:
        print(len(stack))
    elif c == 4:
        print(0 if stack else 1)
    else:
        print(stack[-1] if stack else -1)